features/netsocket/WiFiInterface.h

```diff
@@ -100,6 +100,118 @@ class WiFiInterface: public NetworkInterface
+     *
+     *  This function has to be called before the access point is started.
+     *
+     *  @param gateway     Null-terminated representation of the local gateway
+     *  @param netmask     Null-terminated representation of the network mask
+     *  @return            0 on success, negative error code on failure
+     */
+    virtual nsapi_error_t ap_set_network(const char *gateway, const char *netmask)
+    {
+        return NSAPI_ERROR_UNSUPPORTED;
+    }
+
+    /** Set the WiFi access point network SSID and credentials
+     *
+     *  This function has to be called before the access point is started.
+     *
+     *  @param ssid      Name of the network to create
+     *  @param pass      Security passphrase to protect the network with (Default: No password)
+     *  @param security  Type of encryption for connection (Default: Open network)
+     *  @return          0 on success, or error code on failure
+     */
+    virtual nsapi_error_t ap_set_credentials(const char *ssid, const char *pass = NULL,
+                                             nsapi_security_t security = NSAPI_SECURITY_NONE)
+    {
+        return NSAPI_ERROR_UNSUPPORTED;
+    }
+
+    /** Set the WiFi access point discoverability
+     *
+     *  @param hidden    true to hide the SSID, false to advertise the SSID
+     *  @return          0 on success, or error code on failure
+     */
+    virtual nsapi_error_t ap_set_hidden(bool hidden)
+    {
+        return NSAPI_ERROR_UNSUPPORTED;
+    }
+
+    /** Set the WiFi access point network channel
+     *
+     *  @param channel   Channel on which the network is to be created, or 0 for auto
+     *  @return          0 on success, or error code on failure
+     */
+    virtual nsapi_error_t ap_set_channel(uint8_t channel)
+    {
+        return NSAPI_ERROR_UNSUPPORTED;
+    }
+
+    /** Set up the access point and start accepting connections
+     *
+     *  Before this function is called @a ap_set_credentials and @a ap_set_network needs to be called to set up the
+     *  basic details, like BSSID and IP addresses. Security can be selected using @a ap_set_credentials, defaults to
+     *  open network if not set. Channel of operation can be supplied using @a ap_set_channel, it will be selected
+     *  automatically if not set. BSSID can be hidden using @a ap_set_hidden, it will be discoverable by default.
+     *
+     *  @return         0 on success, or error code on failure
+     */
+    virtual nsapi_error_t ap_start()
+    {
+        return NSAPI_ERROR_UNSUPPORTED;
+    }
+
+    /** Disconnect all clients and stop the access point
+     *
+     *  @return         0 on success, or error code on failure
+     */
+    virtual nsapi_error_t ap_stop()
+    {
+        return NSAPI_ERROR_UNSUPPORTED;
+    }
+
+    /** Return number of connected stations
+     *
+     *  @return         Number of connected WiFi clients, or negative on error
+     */
+    virtual nsapi_size_or_error_t ap_sta_number()
```

Comment:
This should probably have "get" in the name. IMO we should avoid shortened names for clarity unless existing standards exist (ie recv). Also it's not really getting the number of "stations" is it? I might just be confused on the names here.

Maybe `ap_get_client_count()`?